Src/main.c

```diff
@@ -108,6 +109,11 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+    uint16_t adcOutput = 0;
+    adcOutput = HAL_ADC_GetValue(&hadc); //This will give a value corrosponding to 0-4095 and will be between 0-3.3v. It is a 12 bit long int.
+    uint16_t pwmPercentage = adcOutput/4095 * 100;
```

Comment:
Fun fact about this line. No matter the value of adcOutput (so long as it's smaller than 4095), pwmPercentage will always be 0. This is because adcOutput is an integer, so is pwmPercentage , and so are the litterals "4095" and "100". So integer math is done and adcOutput/4095 always evaluates to 0. 

To make something like this work, you'd need to do something more like this: 
`uint16_t pwmPercentage = (uint16_t) ((float) adcOutput / 4095.0f  * 100.0f);`

A little reminder that C will stab you in the back at every opportunity 😁.

Reply:
Haha makes sense! Perfect time to use atoi() :D